src/resolve/resolved-dns-server.c

```diff
@@ -213,14 +213,15 @@ void dns_server_move_back_and_unmark(DnsServer *s) {
         }
 }
```

Comment:
hmm, can you elaborate on the reasoning of this commit? I mean, I'd assume that servers that can do DNS-over-TLS can also do EDNS0+DO+LARGE just fine, hence I always assumed we'd add DNS-over-TLS simply as another feature level on top of all the others?

Can you explain what made you split this up?

Note that the feature level the way it was currently implemented implemented a state machine of sorts, and not just a linear level. Specifically, we'd start out at LARGE, and then when encountering too many failures we'd go to DO, then to EDNS0, then to UDP, then to TCP, and then back to UDP and then TCP again and so on. Or in other words:

```
   LARGE → DO → EDNS0 → UDP → TCP
                         ↑     ↓     
                            ←
```

And I'd always have assumed it would now become:

```
   TLS → LARGE → DO → EDNS0 → UDP → TCP
                               ↑     ↓     
                                  ←
```

Where TLS is supposed to mean TLS-over-TCP, and LARGE, DO, EDN0 all mean extension of plain UDP.